opnode/rollup/derive/payload_attributes.go

```diff
@@ -192,17 +193,18 @@ func BatchesFromEVMTransactions(config *rollup.Config, txs []*types.Transaction)
 			}
 			// some random L1 user might have sent a transaction to our batch inbox, ignore them
 			if seqDataSubmitter != config.BatchSenderAddress {
+				// TODO: log/record metric
 				continue // not an authorized batch submitter, ignore
 			}
 			batches, err := DecodeBatches(config, bytes.NewReader(tx.Data()))
 			if err != nil {
-				// TODO: log error
-				continue
+				// TODO: Maybe log, otherwise this is unrecoverable
+				return nil, err
```

Comment:
This should be an ignored error, that's very important: a single bad tx shouldn't cause us to completely fail to parse the L1 data. If something is invalid, we should make the verifier skip it, and thus the L2 can continue making progress.

Reply:
This is true, but an error in the validation code here will cause a chain split if it's not the same across different rollup node versions.

Reply:
The attack is the sequencer purposely includes a faulty transaction in the batch to halt all of the verifiers. I think adding a comment here and having a `MUST` in the spec is a good idea to ensure that this is taken into account

Reply:
done